packages/core/src/app/customer/CustomerRegistered.test.tsx

```diff
@@ -0,0 +1,302 @@
+                    </AnalyticsProviderMock>
+                </LocaleContext.Provider>
+            </CheckoutProvider>
+        );
+    });
+
+    it('renders login form and continue as customer', async () => {
+        const email = faker.internet.email();
+        const password = faker.internet.password();
+
+        jest.spyOn(checkoutService, 'signInCustomer').mockResolvedValue({} as CheckoutSelectors);
+
+        render(<CustomerTest viewType={CustomerViewType.Login} {...defaultProps} />);
+
+        expect(screen.getByTestId('checkout-customer-returning')).toBeInTheDocument();
+
+        expect(screen.getByLabelText(
+            localeContext.language.translate('customer.email_label'),
+        )).toBeInTheDocument();
+
+        await userEvent.type(screen.getByLabelText(
+            localeContext.language.translate('customer.email_label'),
+        ), email);
+
+        expect(screen.getByLabelText(
+            localeContext.language.translate('customer.password_label'),
+        )).toBeInTheDocument();
+
+        await userEvent.type(screen.getByLabelText(
+            localeContext.language.translate('customer.password_label'),
+        ), password);
+
+        await userEvent.click(screen.getByRole('button', {
+            name: localeContext.language.translate('customer.sign_in_action')
+        }));
+
+        expect(checkoutService.signInCustomer).toHaveBeenCalledWith({
+            email,
+            password,
+        });
+    });
+
+    it('displays error message if invalid email', async () => {
+        const email = faker.internet.email();
+        const password = faker.internet.password();
+
+        jest.spyOn(checkoutService, 'signInCustomer').mockResolvedValue({} as CheckoutSelectors);
+
+        render(<CustomerTest viewType={CustomerViewType.Login} {...defaultProps} />);
+
+        expect(screen.getByTestId('checkout-customer-returning')).toBeInTheDocument();
+
+        const invalidEmail = 'test@test.';
+        const emailField = screen.getByLabelText(localeContext.language.translate('customer.email_label'));
+
+        expect(emailField).toBeInTheDocument();
+        await userEvent.type(emailField, invalidEmail);
+
+        expect(screen.getByLabelText(
+            localeContext.language.translate('customer.password_label'),
+        )).toBeInTheDocument();
+
+        await userEvent.type(screen.getByLabelText(
+            localeContext.language.translate('customer.password_label'),
+        ), password);
+
+        await userEvent.click(screen.getByRole('button', {
+            name: localeContext.language.translate('customer.sign_in_action')
+        }));
+
+        expect(screen.getByText(localeContext.language.translate('customer.email_invalid_error'))).toBeInTheDocument();
+
+        expect(checkoutService.signInCustomer).not.toHaveBeenCalled();
+
+        expect(emailField).toBeInTheDocument();
+        await userEvent.clear(emailField);
+        await userEvent.type(emailField, email);
```

Comment:
Do we need this if we are testing for an invalid email address, think once we assert that an error is displayed the test assert should be ok.